src/providers/arcgisrest/qgsarcgisrestutils.cpp

```diff
@@ -964,6 +982,24 @@ QgsFeatureRenderer *QgsArcGisRestUtils::parseEsriRenderer( const QVariantMap &re
   return nullptr;
 }
 
+QString QgsArcGisRestUtils::parseEsriArcadeExpression( const QString &string )
+{
+  QString expression = string;
+
+  // Replace a few Arcade token to QGIS equivalents
+  expression = expression.replace( QRegularExpression( "(?=([^\"\\\\]*(\\\\.|\"([^\"\\\\]*\\\\.)*[^\"\\\\]*\"))*[^\"]*$)(\\s|^)CONCAT(\\s|$)" ), QString( "\\4||\\5" ) );
+  expression = expression.replace( QRegularExpression( "(?=([^\"\\\\]*(\\\\.|\"([^\"\\\\]*\\\\.)*[^\"\\\\]*\"))*[^\"]*$)(\\s|^)NEWLINE(\\s|$)" ), QString( "\\4'\\n'\\5" ) );
```

Comment:
etc ;)

Reply:
@nyalldawson , yeah, indeed, etc. :smile: I'm not in need of anything else ATM, I'm sure others will eventually expand.